go/tasks/pluginmachinery/core/exec_context.go

```diff
@@ -18,8 +18,11 @@ type TaskReader interface {
 // An interface that is passed to every plugin invocation. It carries all meta and contextual information for the current
 // task execution
 type TaskExecutionContext interface {
-	// Returns a resource manager that can be used to create reservations for limited resources
-	ResourceManager() ResourceManager
+	// During execution time, plugins can call AllocateResource() to register a token to the token pool associated with a resource with the resource manager.
+	AllocateResource(ctx context.Context, namespace ResourceNamespace, allocationToken string, constraintsSpec ResourceConstraintsSpec) (AllocationStatus, error)
```

Comment:
Can you elaborate on why do we need this change? it's sort of a break from the rest of the APIs in this interface...

Reply:
yes these calls are now intercepted (in propeller code) to record allocated resource pool information

Reply:
so Propeller also provides the implementation of the ResourceManager as well.. can't we intercept those calls there instead?

Reply:
but this interface is tied to a specific task execution context. more convenient since there's no global state that needs to be accounted for in tracing resource allocation to specific task executions